src/BTree.mo

```diff
@@ -0,0 +1,110 @@
+/// Imperative sequences as B-Trees.
+
+import A "Array";
+import I "Iter";
+import List "List";
+import Option "Option";
+import Order "Order";
+import P "Prelude";
+import Prim "mo:⛔";
+
+module {
+
+  /// Constants we use to shape the tree.
+  /// See https://en.wikipedia.org/wiki/B-tree#Definition
+  module Constants {
+    let MAX_CHILDREN = 4;
+  };
+
+  public type Compare<K> = {
+    compare : (K, K) -> Order.Order
+  };
+
+  public type Data<K, V> = [(K, V)];
+
+  public type Index<K, V> = {
+    data : Data<K, V>;
+    trees : [Tree<K, V>];
+  };
+
+  public type Tree<K, V> = {
+    #index : Index<K, V>;
+    #data : Data<K, V>;
+  };
+
+  func find_data<K, V>(data : Data<K, V>, find_k : K, c : Compare<K>) : ?V {
```

Comment:
`c` can just be of type `(K, K) -> Order.Order`. 

So the function signature can become 

```
func find_data<K, V>(data : Data<K, V>, find_k : K, compareTo: (K, K) -> Order.Order) : ?V {
```

Reply:
what's the advantage of that?  Less cognitive overhead in some way?

I actually like the record with a function inside pattern, even with just one function: If I want to add debugging later, or any other operations, it helps to have a record in place already to place those operations.

Reply:
Okay.  I've removed that record-of-function from the public-facing API used in tests, etc.

Reply:
FWIW, it was indeed useful to keep in in the checking API (just for tests), since then I could debug by printing out keys from within the BTree code.

Without that record with a `show` function in it, there's no way for the generic code to print its data.